test/connectors/SoftDrinksIndustryLevyConnectorSpec.scala

```diff
@@ -0,0 +1,224 @@
+                                 |  }
+                                 |""".stripMargin)
+  )
+  val mockHttp = mock[HttpClient]
+
+
+  val softDrinksIndustryLevyConnector = new SoftDrinksIndustryLevyConnector(http =mockHttp, config)
+
+  val sdilReturn = SdilReturn(
+    (1L, 1L),
+    (1L,1L),
+    List(SmallProducer("test", "test", (15,14))),
+    (1L, 1L),
+    (1L, 1L),
+    (1L, 1L),
+    (1L, 1L)
+    //submittedOn: Option[LocalDateTime] = None
+  )
+
+  val revisedSdilReturn = SdilReturn(
+    (2L, 2L),
+    (2L,2L),
+    List(SmallProducer("test2", "test2", (16,15))),
+    (2L, 2L),
+    (2L, 2L),
+    (2L, 2L),
+    (2L, 2L)
+    //submittedOn: Option[LocalDateTime] = None
+  )
+
+  val aSubscription = RetrievedSubscription(
+    "0000000022",
+    "XKSDIL000000022",
+    "Super Lemonade Plc",
+    UkAddress(List("63 Clifton Roundabout", "Worcester"), "WR53 7CX"),
+    RetrievedActivity(false, true, false, false, false),
+    LocalDate.of(2018, 4, 19),
+    List(
+      Site(
+        UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP"),
+        Some("88"),
+        Some("Wild Lemonade Group"),
+        Some(LocalDate.of(2018, 2, 26))),
+      Site(
+        UkAddress(List("117 Jerusalem Court", "St Albans"), "AL10 3UJ"),
+        Some("87"),
+        Some("Highly Addictive Drinks Plc"),
+        Some(LocalDate.of(2019, 8, 19))),
+      Site(
+        UkAddress(List("87B North Liddle Street", "Guildford"), "GU34 7CM"),
+        Some("94"),
+        Some("Monster Bottle Ltd"),
+        Some(LocalDate.of(2017, 9, 23))),
+      Site(
+        UkAddress(List("122 Dinsdale Crescent", "Romford"), "RM95 8FQ"),
+        Some("27"),
+        Some("Super Lemonade Group"),
+        Some(LocalDate.of(2017, 4, 23))),
+      Site(
+        UkAddress(List("105B Godfrey Marchant Grove", "Guildford"), "GU14 8NL"),
+        Some("96"),
+        Some("Star Products Ltd"),
+        Some(LocalDate.of(2017, 2, 11)))
+    ),
+    List(),
+    Contact(Some("Ava Adams"), Some("Chief Infrastructure Agent"), "04495 206189", "Adeline.Greene@gmail.com"),
+    None
+  )
+
+  val emptySdilReturn = SdilReturn((0L,0L),(0L, 0L),List.empty,(100L, 100L),(0L,0L),(0L,0L),(0L,0L))
+
+  val returnPeriod = ReturnPeriod(year = 2022, quarter = 3)
+  val sdilNumber: String = "XKSDIL000000022"
+  val utr: String = "1234567891"
+  val returnPeriodList = List(ReturnPeriod(year = 2022, quarter = 3), ReturnPeriod(year = 2021, quarter = 3), ReturnPeriod(year = 2020, quarter = 3))
+
+  implicit val hc = HeaderCarrier()
+
+  "SoftDrinksIndustryLevyConnector" must {
+
+    "return a subscription Successfully" in {
+
+      val identifierType: String = "0000000022"
+
+      when(mockHttp.GET[Option[RetrievedSubscription]](any(), any(), any())(any(), any(), any())).thenReturn(Future.successful(Some(aSubscription)))
+      Await.result(softDrinksIndustryLevyConnector.retrieveSubscription(sdilNumber,identifierType), 4.seconds) mustBe  Some(aSubscription)
```

Comment:
Use whenReady blocks instead of awaiting these, search returns frontend for examples. Same comment for the rest of the awaits.

Reply:
completed